### DISCUSSION TITLE
Cotton model in APSIM Next Gen

### DISCUSSION THREAD

**Discussion by mccarthyalison** [APSIMInitiative/ApsimX#9527] [General]
Good morning

I am wondering how close the cotton model is to being completed in APSIM Next Gen? I see some work was done on the prototype three weeks ago and so it is still being actively developed but has not been through peer review. I have previously used the cotton model in APSIM Classic for simulating impacts on yield from irrigation management and am interested in transferring this to APSIM Next Gen - do you anticipate the model prototype would be accurate enough for this application please?

Regards
Alison

**Comment by wjc118** [comment]
Agree with your idea, I also hope that the cotton module can be developed as soon as possible

**Comment by Yakup63563** [comment]
I will conduct research on cotton. I'm waiting for it.

**Comment by MuhammadZeeshanHussain** [comment]
I am new to here, I am interested to add new cultivar file in Apsim, but I am unable to findout cotton model in APSIM next generation. Is there anyone who can help me. Thansks zeeshanawan1919@gmail.com

**Reply by hut104** [reply]
Unfortunately the cotton model for APSIM Next Generation is still under development.

**Comment by MuhammadZeeshanHussain** [comment]
When I add an "ini" file in cotto simulation, APSIM version 7.10. I get an error in paddok component. "APSIM Fatal Error ------------------- Exception has been thrown by the target of an invocation. Component name: paddock !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!". I cannot add ini file in my simulation, because I have to introduce new cultivars, so anyone can help me. zeeshanawan1919@gmail.com kindly help me, thanks

**Comment by MuhammadZeeshanHussain** [comment]
I am working on the APSIM 7.10 r4220 version. But, when I tried to add 'ini
file' it showed an error. Actually, I want to introduce new cultivars in my
simulation and especially the cotton crop, it shows error. Can you help me
or guide me? Attached summary of error.

Thanks and regards
Muhammad Zeeshan Hussain
M.Sc. (Hons.) Agronomy

**Comment by peter-devoil** [comment]
Muhammad - [this q in the classic repository](https://github.com/APSIMInitiative/APSIM710/issues/2241) might help you, if you havent already solved it.

**Comment by MuhammadZeeshanHussain** [comment]
Dear Peter, Thanks alot for guiding me.

**Comment by anyone-wrt** [comment]
Dear APSIM Initiative Team,

I am a researcher currently using APSIM 7.10 (Classic) for cotton crop modeling. I am writing to inquire about the current development status and the expected release timeline for the Cotton module in APSIM Next Generation.

While APSIM 7.10 has been a robust tool, I have encountered significant limitations in simulating modern cotton management practices, particularly regarding Topping (Manual and Chemical/DPC).

In my current research, I find that:

Source-Sink Dynamics: The current OZCOT module in 7.10 lacks a flexible mechanism to simulate the active redistribution of carbohydrates from stems to bolls following topping.

Growth Regulation: The rigid structure of the old module makes it difficult to model the physiological compensation effects (e.g., changes in RUE or specific partitioning coefficients) after applying growth retardants like DPC.

I am eager to transition to APSIM Next Gen because its Plant Modelling Framework (PMF) appears much more capable of handling these complex physiological feedbacks through dynamic scripts.

Could you please provide information on:

Is there an estimated timeframe for a stable Cotton PMF release?

If a full version is not yet available, is there a beta version or a generic "Plant2" template specifically calibrated for cotton that I could test or contribute to?

**Reply by ric394** [reply]
Hi @anyone-wrt,

1) The Next Gen cotton model is currently in active development and frequently being updated, so I would not recommend anyone try and use it currently. For that reason, the latest prototype is currently not available outside the project developers.

2) The project developing this model is looking to have the new model reviewed by the APSIM reference panel in the next few months, so an official release will come after any changes from that review.

**Reply by anyone-wrt** [reply]
Hi @ric394,Thanks for the heads-up! I will stick with APSIM 7.10 for my current study.On that note, I am trying to simulate manual and chemical topping (DPC) to see their effects on "concentrated maturity" in Southern Xinjiang. In the Classic version, is there a recommended way to "stop" the main stem growth in the Manager script?Specifically, I need to redistribute dry matter to bolls once the terminal bud is removed or inhibited. Any tips or existing script examples for Classic cotton would be immensely helpful.
